scanpy/plotting/_tools/scatterplots.py

```diff
@@ -1256,8 +1257,10 @@ def _color_vector(
     }
     # If color_map does not have unique values, this can be slow as the
     # result is not categorical
-    color_vector = pd.Categorical(values.map(color_map, na_action="ignore"))
-
+    if Version(pd.__version__) < Version("2.1.0"):
```

Comment:
```suggestion
    if Version(version("pandas")) < Version("2.1.0"):
```

see https://github.com/scverse/anndata/pull/1314#discussion_r1453509405